src/main/java/org/folio/services/transactions/BatchTransactionService.java

```diff
@@ -70,4 +80,35 @@ private Future<Void> updateTransaction(Transaction transaction, RequestContext r
       .withTransactionsToUpdate(singletonList(transaction));
     return processBatch(batch, requestContext);
   }
+
+  private Future<Void> checkDeletions(Batch batch, RequestContext requestContext) {
+    if (batch.getIdsOfTransactionsToDelete().isEmpty()) {
+      return succeededFuture();
+    }
+    return anyConnectedToInvoice(batch.getIdsOfTransactionsToDelete(), requestContext)
+      .map(connected -> {
+        if (FALSE.equals(connected)) {
+          return null;
+        }
+        logger.warn("validateDeletion:: Tried to delete transactions but one is connected to an invoice, ids={}",
+          batch.getIdsOfTransactionsToDelete());
+        throw new HttpException(422, DELETE_CONNECTED_TO_INVOICE.toError());
+      });
+  }
+
+  private Future<Boolean> anyConnectedToInvoice(List<String> ids, RequestContext requestContext) {
+    // We want to know if the order with the given encumbrance is connected to an invoice.
+    // To avoid adding a dependency to mod-invoice, we check if there is a related awaitingPayment transaction
```

Comment:
Please add details about this into related story MODFISTO-432(or create new one)
- this check was not added to mod-finance-storage because we have an encumbrances script
- this can be reproducible by flow: open order, approve invoice, unopen order, remove fund distrib line

When we have some bugs - we trying to find story number by Annotate functions to know more about the requirements for particular change. It can be useful for future

Reply:
I created [MODORDERS-1043](https://folio-org.atlassian.net/browse/MODORDERS-1043) about this case, and added an implementation note in [MODFISTO-432](https://folio-org.atlassian.net/browse/MODFISTO-432).